### ISSUE TITLE
Won't parse

### ISSUE THREAD

**Issue by 737simpilot** [CIDRAM/CIDRAM#262]
I just created two Cloudflare blocks, added them to the IPv4 and IPv6 lists and I tested the new blocks by checking IPs in the test page, nothing shows up as blocked. I then added random IPs in these files for testing to see if they would parse and show up in the test page. Still nothing. I have done this before and I don't think I'm missing a step. I already have my own country block lists. 

Another issue- For testing and to rule out my configuration, I installed the latest plain vanilla CIDRAM, but here too I have an issue. I can't log in! I renamed the config.ini file, saw nothing in this new ini file that turns off the frontend, and when I go to login I get a blank white page. 

Attached are the blocks I created. I don't know what tag is, but it looks important and I have been adding that to my custom signatures. 

Reason why I created these blocks is that I saw other Cloudflare IPs hit me that weren't blocked by Macmathan's lists. He wanted me to contact Cloudflare on their published list. Not doing that. Some of these IPs seem to belong to incognito stuff for Apple. Something like iCloud privacy or whatever...

[cloudflare_blocks.zip](https://github.com/CIDRAM/CIDRAM/files/7232402/cloudflare_blocks.zip)

**Comment by Maikuolan** [comment]
> I just created two Cloudflare blocks, added them to the IPv4 and IPv6 lists and I tested the new blocks by checking IPs in the test page, nothing shows up as blocked. I then added random IPs in these files for testing to see if they would parse and show up in the test page. Still nothing. I have done this before and I don't think I'm missing a step. I already have my own country block lists.

The "Deny" keyword is case-sensitive (currently, at least; I could investigate the possibility of changing it, but I'm not sure offhandedly whether changing it could be problematic; I'd need to get back to you about that). Currently, in your attached files, you've included it like "DENY" (all upper-case). Try changing it from "DENY" to "Deny", and giving it another go. Let me know how it goes. :-)

> Another issue- For testing and to rule out my configuration, I installed the latest plain vanilla CIDRAM, but here too I have an issue. I can't log in! I renamed the config.ini file, saw nothing in this new ini file that turns off the frontend, and when I go to login I get a blank white page.

Have seen this issue only just now, so haven't looked into that yet (figured I may as well reply immediately with what I could ascertain by glancing at the attached files to get that out of the way), but I'll look into that now. Cheers.

> I don't know what tag is, but it looks important and I have been adding that to my custom signatures.

"Tag" basically provides a way for us to attach a "name" to a cluster (or "section") of signatures within a signature file. When, for example, you add a signature section to your ignores file, or you visit the "sections list" page at the front-end, or it displays a name for the network being blocked at the Access Denied page, that all comes from the "Tag". In the absence of a tag, CIDRAM will simply fall back to "IPv4" or "IPv6" as a "name" (depending on whether the signatures are IPv4 or IPv6 signatures), so omission of a tag won't break anything (meaning, it isn't strictly necessary). But, I do recommend including them, because having all CIDRAM's signatures be "named" will make things much easier if you need to analyse your logs and diagnose which signatures have caused some particular block events at some point in the future (because it's easier to locate signatures based on actual names than it is to dig through everything based on just the fallback names), and it'll also be easier to then integrate your newly created signatures with auxiliary rules and so on (if you choose to do that at some point).

**Comment by Maikuolan** [comment]
> Have seen this issue only just now, so haven't looked into that yet (figured I may as well reply immediately with what I could ascertain by glancing at the attached files to get that out of the way), but I'll look into that now. Cheers.

Following up on.. Just tried a fresh install at my dev machine, vanilla, front-end enabled, accessed the loader.. Login page displayed correctly. Tried logging in.. Logged in successfully.

Might be a problem with permissions, maybe? Did you check the CHMOD values for the new install when you tried it?

**Comment by 737simpilot** [comment]
Okay, looks like PEBCAK. The DENY thing was correct. My other signatures just have **Deny** and not **DENY**.  Changing it to Deny solved the issue. 

And for some reason when I installed a test plain vanilla CIDRAM version, I could not find disable_frontend in the ini file.  I knew it was there before, but this time I couldn't find it and assumed something had changed. Now I see it and set it to false.

Problem solved. 

![image](https://i.postimg.cc/xTzsLws9/ezgif-com-gif-maker-3.gif)
